### DISCUSSION TITLE
Deprecated API marked for removal, @SuppressWarning and terminalDeprecation set to error

### DISCUSSION THREAD

**Discussion by trancexpress** [eclipse-jdt/eclipse.jdt.core#2500] [Q&A] [answered]
Hi all,

I have a question regarding terminally deprecated API and `org.eclipse.jdt.core.compiler.problem.terminalDeprecation=error`.

In particular, the code base I work on ran into this JDK change: https://bugs.openjdk.org/browse/JDK-8276748

Finalization is now marked for removal, its not just deprecated. In our code base, we use `terminalDeprecation` set to `error`, i.e. we now see compile errors here and there, where we used `finalize()` and `runFinalization()`.

The suggestion in the JDK ticket, to deal with the incompatibility introduced by it, is to use `@SuppressWarning("removal")`. This only suppresses a warning; in a `javac` build, `-Werror` is (I think?) used to cause the deprecated for removal API to be treated as an error. So, `javac`sees the used deprecated API, the warning is suppressed and `-Werror` doesn't produce an error.

In `ecj`, there is the compiler setting above. A `@SuppressWarning` does not influence the compile error produced by the now marked-for-removal API. So when using `ecj`, I can't follow the JDK ticket recommendation on how to deal with the problem, without (yet) doing a code clean-up.

Is there a way to suppress this compile error in `ecj`, only for use of finalization API?

Best regards and thanks,
Simeon

**Comment by stephan-herrmann** [comment, marked as answer]
Have you seen the option **Java Compiler > Errors/Warnings > Annotations ... Suppress optional errors with \@SuppressWarnings**

Is it checked?

**Reply by trancexpress** [reply]
> Have you seen the option **Java Compiler > Errors/Warnings > Annotations ... Suppress optional errors with @SuppressWarnings**
> 
> Is it checked?

It wasn't checked. When using this, the build does go through.

Its a bit more complicated in our case, we must also set the "unnecessary annotation" to a warning - we have 2 builds in this case. One with Java 17, that we use. And one with Java 21 as test, to ensure smooth transition to JLS 21 (which is why we ran into the issue in the first place). So the normal build complains about the unused `\@SuppressWarning("removal")`, but this is out of scope here.

Thank you @stephan-herrmann !
